### PR TITLE
Don't allow page numbers too large when paginating ES results for the API

### LINKED ISSUE #4702: Internal Server Error's
### Describe the problem and steps to reproduce it:
Access the following URLs with a mobile user agent like "Mozilla/5.0 (Android 4.4; Mobile; rv:41.0) Gecko/41.0 Firefox/41.0":

- https://addons-dev.allizom.org/api/v3/accounts/account/6362937?query=ZAP%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%25n%25s%0A
 - https://addons-dev.allizom.org/api/v3/accounts/account/6362937?query=14816458835629746137950925770871450099358545
 - https://addons-dev.allizom.org/api/v3/addons/search/?app=firefox&page_size=4&sort=rating&type=extension&page=42534372092246088007922959442281889602546449&lang=en-US

### What happened?
Server returned:

> {"detail":"Internal Server Error"}

### What did you expect to happen?
An error which indicates the relevant parameter does not have a valid value

### PR DESCRIPTION
**Pull request by diox**
Fix #4702

### CODE REVIEW

File: src/olympia/api/paginator.py [hunk 1]
```diff
@@ -21,6 +21,8 @@ def validate_number(self, number):
             raise PageNotAnInteger('That page number is not an integer')
         if number < 1:
             raise EmptyPage('That page number is less than 1')
+        elif number > 100000:
```
Comment (EnTeQuAk) [inline src/olympia/api/paginator.py]: This should be either `sys.max_int` or 2147483647 (2^^32 - 1) directly which is the limit of the used `integer` type in ElasticSearch. At least that's what the actual exception is all about.

Now that'd be the proper technical fix, I doubt that there'll ever be more than 100k pages unless you own a bulletin board with a "the longest thread of the world"-thread so I'm fine with the 100.000 limit. Let's just put this as a constant at the top of the file for better visibility.

Comment (EnTeQuAk) [inline src/olympia/api/paginator.py]: Actually, reading https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html and search for "index.max_result_window" seems to be a more realistic limit that got implemented in ES 2.x as far as I can see? Maybe we should use that instead?

Comment (diox) [inline src/olympia/api/paginator.py]: I tried `sys.maxint` and it didn't work. First, it's actually much larger than `2^^32 - 1` (`9223372036854775807` on my machine), and second, even `2^^32 - 1` didn't work either, it failed before that. Which is why I settled on a lower, more closer than the realistic limit we'll ever need, value.

Comment (diox) [inline src/olympia/api/paginator.py]: `index.max_result_window` is interesting but does not exist on the version we're using, though.

**Comment by EnTeQuAk** [comment]
Indeed it doesn't but maybe we should use that to be future proof :) 





On Mon, Feb 20, 2017, at 12:10 PM, Mathieu Pillard wrote:

> *@diox* commented on this pull request.



> 

> In src/olympia/api/paginator.py[1]:


> > @@ -21,6 +21,8 @@ def validate_number(self, number): raise
> > PageNotAnInteger('That page number is not an integer') if number <
> > 1: raise EmptyPage('That page number is less than 1') +        elif
> > number > 100000:
>
> index.max_result_window is interesting but does not exist on the
> version we're using, though.
> — You are receiving this because your review was requested. Reply to
> this email directly, view it on GitHub[2], or mute the thread[3].
> 


Links:

  1. https://github.com/mozilla/addons-server/pull/4704#discussion_r101994295
  2. https://github.com/mozilla/addons-server/pull/4704#discussion_r101994295

**Comment by diox** [comment]
https://github.com/mozilla/addons-server/pull/4704/commits/aff92bc15e1533ceb334760a01c48529803b8383

**Pull request merged**